tests/integration/test_refreshable_mv/test.py

```diff
@@ -271,3 +271,28 @@ def test_refresh_vs_shutdown_smoke(started_cluster):
     node1.start_clickhouse()
     node1.query("drop database re sync")
     node2.query("drop database re sync")
+
+
+def test_adding_replica(started_cluster):
+    node1.query(
+        "create database re engine = Replicated('/test/re', 'shard1', 'r1');"
+        "create materialized view re.a refresh every 1 second (x Int64) engine ReplicatedMergeTree order by x as select number*10 as x from numbers(2);"
+        "system wait view re.a")
+    assert node1.query("select * from re.a order by all") == "0\n10\n"
+    assert node1.query("select last_refresh_replica from system.view_refreshes") == "1\n"
+
+    r = node2.query(
+        "create database re engine = Replicated('/test/re', 'shard1', 'r2');"
+        "system sync database replica re")
+    assert node2.query("select * from re.a order by all") == "0\n10\n"
+
+    node1.query("system stop view re.a")
+    r = node2.query(
+        "system wait view re.a;"
+        "system refresh view re.a;"
+        "system wait view re.a;"
```

Comment:
Maybe add a comment on why this should hang without this patch (wait ensures that explicit REFRESH will finish, even in case of disabled RMV)?